src/cargo/util/progress.rs

```diff
@@ -74,6 +74,113 @@ struct Format {
     style: ProgressStyle,
     max_width: usize,
     max_print: usize,
+    taskbar: TaskbarProgress,
+}
+
+/// Taskbar progressbar
+///
+/// Outputs ANSI codes according to the `Operating system commands`.
+/// Currently only works in the Windows Terminal and ConEmu.
+struct TaskbarProgress {
+    show: bool,
+    error: bool,
+}
+
+/// A taskbar progress value printable as ANSI OSC escape code
+enum TaskbarValue {
+    /// Do not output anything
+    None,
+    /// Remove progress
+    Remove,
+    /// Progress value 0-100
+    Value(f64),
+    /// Indeterminate state (no bar, just animation)
+    Indeterminate,
+    Error,
+}
```

Comment:
This is treating error and value is mutually exclusive.  Is that a limitation of the escape codes?

Reply:
No there is no limitation, again your remark was hot to the point!

https://github.com/user-attachments/assets/d07325a5-bc21-4259-b90b-4dff2c80d535